src/utilities/Dealer.cpp

```diff
@@ -19,93 +23,126 @@ Dealer::DealerFlag::DealerFlag(DealerFlagTitle title, DealerFlagPriority priorit
+    std::vector<int> indexRoles;
+    std::vector<int> indexID;
+    std::vector<std::string> roleNames;
+    distribute_init(indexRoles,indexID,roleNames,flagMap);
+
+            // Loop through the order of role priorities (column)
+    for (int i = sizeof(DealerFlagPriority); i >= 0; i--){
+        DealerDistribute current{};
+        // Check if a column has the looked for priorities
+        for (int j = 0; j < scores.size(); j++){
+            // if so, add it to a list and save the index
+            if (scores.at(j).priority == i) {
+                current.currentScores.push_back(scores.at(j).robotScores);    // get the score column
+                current.currentRoles.push_back(j);                      // get the current index
+                current.originalRolesIndex.push_back(indexRoles[j]);    // get the role number
+            }
+        }
+        if (current.currentRoles.size() > 0) {
+            // Return best assignment for those roles (column)
+            rtt::Hungarian::Solve(current.currentScores, current.newAssignments);
+            if (current.newAssignments.size() > 0) {
+                for (int j = 0; j < current.newAssignments.size(); j++) {
+                    if (current.newAssignments[j] >= 0) {
+                        current.currentIDs.push_back(current.newAssignments[j]);                    // get newly assigned robot from current index
+                        current.originalIDsIndex.push_back(indexID[current.currentIDs.back()]);     // get robot number
+                        output.insert({roleNames[current.originalRolesIndex[j]], allRobots[current.originalIDsIndex.back()]});
+                    }
+                }
+                if (output.size() == allRobots.size()) return output;               // case if there are less then 11 bots to distribute
+                distribute_remove(current,indexRoles,indexID,scores);
+            }
+        }
+    }
+    return output;
 }
 
-/* assignment now has the robot index in allRobots (not id) at the role index, and is ordered according to the roleNames
- * for example: assignment[0] = 2 // index
- * and roleNames[0] = role_1
- * robot_id = allRobots[index]
- * --> we can therefore make a map of <rolename, robot_id>
- */
-std::unordered_map<std::string, v::RobotView> Dealer::mapFromAssignments(const std::vector<v::RobotView> &allRobots, const Dealer::FlagMap &flagMap,
-                                                                         const std::vector<int> &assignment) const {
-    vector<string> orderedRoleNames;
+void Dealer::distribute_init(std::vector<int>& indexRoles, std::vector<int>& indexID, std::vector<std::string>& roleNames, const Dealer::FlagMap &flagMap) {
+    indexRoles.reserve(flagMap.size());
+    indexID.reserve(flagMap.size());
+    roleNames.reserve(flagMap.size());
+    for (int i = 0; i < flagMap.size(); i++) {
+        indexRoles.push_back(i);
+        indexID.push_back(i);
+    }
```

Comment:
This reads a bit like a constructor, maybe consider making a data format for this (but this is fine, too).

Reply:
Is there something as to many structs? haha
I couldnt find a way (but i am sure there is) to get a incrementing vector constructed..

Reply:
std::iota does exactly what you need (#include <numeric>). 

You could simply write:

indexRoles.resize(flagMap.size());
std::iota(indexRoles.begin(),indexRoles.end(),0);

But this is fine too (it's clear enough)

Reply:
Too many structs can be a thing, but typically encapsulation helps make your code more readable and easier to argue about.